### PR TITLE
Migration of claims flow from Odyssey to client (UI Only)

### PR DESCRIPTION
**Pull request by juliaendre**
## [CLX-426]

Migration of claims flow from Odyssey to client (UI Only).

- Added screens for 
- [CLX-411]**: Contact information screen**
SubmitClaimContactScreen

- [CLX-380]**: Incident info screen** 
SubmitClaimOccurrenceScreen
DatePickerScreen
LocationPickerScreen
 
- [CLX-384]**: Product info screen**
SubmitClaimObjectInformation
DamagePickerScreen
ModelPickerScreen

- [CLX-385]**: Summary screen**
SubmitClaimSummaryScreen
SubmitClaimEditSummaryScreen

- [CLX-386]**: Audio recorder screen**
SubmitClaimAudioRecordingScreen
SubmitClaimSuccessScreen

- [CLX-388]**: Checkout progress**
SubmitClaimCheckoutTransferringScreen
SubmitClaimCheckoutTransferringDoneScreen

- [CLX-387]**: Self checkout**
SubmitClaimCheckoutNoRepairScreen

Added states in **ClaimsState** and Journeys in **ClaimsJourney** . This  code part can probably be improved.

- [CLX-426]**: Added new feature flag claims_flow from unleash**

## Checklist

- [X] Dark/light mode verification
- [X] Landscape verification
- [X] iPad verification
- [X] iPod/iPhone 12 mini/iPhone SE verification

[CLX-426]: https://hedvig.atlassian.net/browse/CLX-426?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CLX-411]: https://hedvig.atlassian.net/browse/CLX-411?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CLX-380]: https://hedvig.atlassian.net/browse/CLX-380?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CLX-384]: https://hedvig.atlassian.net/browse/CLX-384?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CLX-385]: https://hedvig.atlassian.net/browse/CLX-385?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CLX-386]: https://hedvig.atlassian.net/browse/CLX-386?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CLX-388]: https://hedvig.atlassian.net/browse/CLX-388?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CLX-387]: https://hedvig.atlassian.net/browse/CLX-387?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift [hunk 1]
```diff
@@ -0,0 +1,353 @@
+import AVFAudio
+import Combine
+import Foundation
+import SwiftUI
+import hAnalytics
+import hCore
+import hCoreUI
+import hGraphQL
+
+struct EmbarkRecorderTracking {
```
Comment (sampettersson) [inline Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift:10]: This should probably not be called `Embark*`, Embark is our old flow system.

File: Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+
+        }
+        .hFormAttachToBottom {
+
+            ZStack(alignment: .bottom) {
+
+                if let recording = audioRecorder.recording {
+                    VStack(spacing: 12) {
+
+                        /* TODO: Change the audioplayer to take audioplayer as an input */
```
Comment (sampettersson) [inline Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift:130]: You left a TODO here

File: Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+                        TrackPlayer(audioPlayer: audioPlayer)  // {
```
Comment (sampettersson) [inline Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift:131]: Some left comments

File: Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+                        hButton.LargeButtonText {
+                            tracking.onRetry.send()
+                            withAnimation(.spring()) {
+                                audioRecorder.restart()
+                            }
+                        } content: {
+                            hText(L10n.embarkRecordAgain)
+                        }
+                    }
+                    //                    }
```
Comment (sampettersson) [inline Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift:153]: Left comment

File: Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift [hunk 5]
```diff
@@ -0,0 +1,353 @@
+import AVFAudio
```
Comment (sampettersson) [inline Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift:1]: I would try to split this screen up, a lot of code in the same file.

File: Projects/Claims/Sources/Views/SubmitClaim/SummaryInfo/SubmitClaimEditSummaryScreen.swift [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+                                    .foregroundColor(hLabelColor.primary).colorScheme(.light)
+                                    .padding([.top, .bottom], 11)
+                                    .padding([.trailing, .leading], 12)
+                            }
+                            .background(hGrayscaleColor.one)
+                            .cornerRadius(.defaultCornerRadius)
+                        }
+                        .frame(maxWidth: .infinity, alignment: .leading)
+                    }
+                    .padding([.leading, .trailing], -16) /* TODO: Possible to make this better? */
```
Comment (sampettersson) [inline Projects/Claims/Sources/Views/SubmitClaim/SummaryInfo/SubmitClaimEditSummaryScreen.swift:45]: Why did you do this? 🤔

File: Projects/Claims/Sources/Views/submitClaim/Checkout/SubmitClaimCheckoutTransferringScreen.swift [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+    ) {
+        self.content = content
+    }
+
+    @hColorBuilder
+    var largeCircleColor: some hColor {
+        if colorScheme == .dark {
+            hTintColor.yellowOne.opacity(0.25)
+        } else {
+            //            hTintColor.yellowThree
```
Comment (sampettersson) [inline Projects/Claims/Sources/Views/submitClaim/Checkout/SubmitClaimCheckoutTransferringScreen.swift:96]: Left comment

File: Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift (hunk 1 above)
Comment (julia-andersson) [reply Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift:10]: Agree, this was a mistake

File: Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift (hunk 5 above)
Comment (julia-andersson) [reply Projects/Claims/Sources/Views/SubmitClaim/SubmitClaimAudioRecordingScreen.swift:1]: Will fix

**Pull request merged**